src/camera.rs

```diff
@@ -1,43 +1,89 @@
 //! Contains useful [`Camera`](struct.Camera.html) struct and `Projections`.
 use std::ops;
 
-use cgmath::{ortho as cgmath_ortho, perspective as cgmath_perspective, Deg};
+use cgmath;
 use mint;
 
 use NodePointer;
 use object::Object;
 
+/// A camera's projection.
+pub enum Projection {
+    /// An orthographic projection.
+    Orthographic(Orthographic),
+
+    /// A perspective projection.
+    Perspective(Perspective),
+}
+
 /// Camera is used to render Scene with specific `Projection`.
-pub struct Camera<P> {
+pub struct Camera {
     pub(crate) object: Object,
+
     /// Projection parameters of this camera.
-    pub projection: P,
+    pub projection: Projection,
 }
 
-impl<P> AsRef<NodePointer> for Camera<P> {
+impl Camera {
+    /// Computes the projection matrix representing the camera's projection.
+    pub fn project(&self, aspect_ratio: f32) -> mint::ColumnMatrix4<f32> {
```

Comment:
Actually, `get_matrix` is more obvious for me. Any reason for that naming?

Reply:
I renamed once and forgot to rename again. `Camera::matrix` is probably a better name, would you agree?

Reply:
Or even, `Camera::compute_projection_matrix`.

Reply:
I prefer `matrix` or `projection_matrix`